lib/biker.rb

```diff
@@ -1,3 +1,22 @@
 class Biker
+attr_reader :name,
+            :max_distance            
 
+attr_accessor :rides,
+              :acceptable_terrain
+
+  def initialize(name, max_distance)
+    @name = name
+    @max_distance = max_distance
+    @rides = {}
+    @acceptable_terrain = []
+  end
+
+  def learn_terrain!(terrain)
+    @acceptable_terrain << terrain
+  end
+
+  def log_ride(ride, time)
+    @rides.merge {|ride, time| [time] << time}
```

Comment:
You were on the right track, but next time, remember that you can go back to the basics rather than trying to use a more unfamiliar method like `merge`. It's totally fine to add an if statement here to avoid the nil errors!